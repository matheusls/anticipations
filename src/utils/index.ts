export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
