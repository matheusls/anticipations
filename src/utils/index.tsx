import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
