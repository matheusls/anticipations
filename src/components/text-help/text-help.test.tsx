import { render, screen } from 'tests/utils';

import TextHelp from './text-help';

describe('<TextHelp />', () => {
  it('should match snapshot', () => {
    const { container } = render(<TextHelp>TextHelp component</TextHelp>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render TextHelp correctly', () => {
    render(<TextHelp>TextHelp component</TextHelp>);

    expect(screen.getByText(/texthelp component/i)).toBeInTheDocument();
  });
});
