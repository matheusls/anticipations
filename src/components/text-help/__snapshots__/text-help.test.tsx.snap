// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextHelp /> should match snapshot 1`] = `
.c0 {
  color: #cecece;
  font-size: 0.688rem;
  font-weight: 700;
  margin: 0;
  margin-top: 3px;
}

<p
  class="c0"
>
  TextHelp component
</p>
`;
